### PR TITLE
Rename edit-post__fade-in-animation and unify keyframe definitions

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Extracted from #65203

Rename the `edit-post__fade-in-animation` to a more generic name (deprecating the previous name), unify underlying keyframe definitions to be under the same package, and add an equivalent "fade out" mixin.

## Why?

See https://github.com/WordPress/gutenberg/pull/64580#discussion_r1751728630

## How?

- renamed `edit-post__fade-in-animation` to `animation__fade-in`, deprecated old name
- moved fade in keyframes definition to the same package as the animation mixin (yes, there could be duplication in the output — but IMO it's better to trade off some redundancy if we can get rid off circular dependencies & we don't get the animation keyframes defined across multiple packages.
- added the fade out equivalent, `animation__fade-out`

## Testing Instructions

Make sure that the `Modal` component's enter animation (fade in) works as on trunk, both in Storybook and across various parts of Gutenberg.

### CODE REVIEW

File: packages/edit-post/src/style.scss [hunk 1]
```diff
@@ -3,22 +3,6 @@
 @import "./components/meta-boxes/meta-boxes-area/style.scss";
 @import "./components/welcome-guide/style.scss";
 
-/**
- * Animations
- */
-
-// These keyframes should not be part of the _animations.scss mixins file.
-// Because keyframe animations can't be defined as mixins properly, they are duplicated.
-// Since hey are intended only for the editor, we add them here instead.
```
Comment (ciampo) [inline packages/edit-post/src/style.scss:12]: With the changes introduced by this PR, there may be some duplication — but IMO it's a much better compromise than all the potential downsides listed in https://github.com/WordPress/gutenberg/pull/64580#discussion_r1751728630

File: packages/base-styles/_animations.scss [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+		}
+		to {
+			opacity: 1;
+		}
+	}
+
+
+	animation: __wp-base-styles-fade-in $speed linear $delay;
+	animation-fill-mode: forwards;
+	@include reduce-motion("animation");
```
Comment (mirka) [inline packages/base-styles/_animations.scss:20]: What are your thoughts on the continued usage of this mixin? For new code, should we just write out the media query like we're doing in the components package?

**Review by mirka — APPROVED** [review]
Looks good 👍

File: packages/base-styles/_animations.scss (hunk 2 above)
Comment (ciampo) [reply packages/base-styles/_animations.scss:20]: I like the approach that we're taking in components package, ie. "only add animation styles if the user does NOT prefer reduced motion".

Having said that, I would prefer not to make such a change here for two reasons:
- consistency: if make such a change, we could consider doing so at the project level?
- the `reduce-motion` mixin sets durations to `0s`, which is technically different from not having an animation at all. For example, I'm not sure how animation and transition events would fire across browsers, and that could introduce unexpected behaviors.

To sum it up: I'd be up for making the change, but I'd make it separately from this PR and gradually:
- refactor usages one by one to `@media not (prefers-reduce-motion) { /* animation code /* }`, test for breaking changes
- when all usages are migrated, deprecate the mixing

Comment (DaniGuardiola) [reply packages/base-styles/_animations.scss:20]: agree with @ciampo, worth doing but with precaution

Comment (mirka) [reply packages/base-styles/_animations.scss:20]: Got it 👍

Comment (DaniGuardiola) [reply packages/base-styles/_animations.scss:20]: That said, regarding this:

> the reduce-motion mixin sets durations to 0s, which is technically different from not having an animation at all. For example, I'm not sure how animation and transition events would fire across browsers, and that could introduce unexpected behaviors

I'm not 100% sure, but I can't see how a duration of 0 would be effectively any different from disabling the animation itself, other than if there's a delay in one of them or something like that which I doubt. Btw, now that I think about it that approach (setting duration to 0) was flawed in that way. A delay would still affect it. One more reason to migrate away from that pattern.

Comment (mirka) [reply packages/base-styles/_animations.scss:20]: FWIW the duration is set to 1ms, which may be even more different 😅
